app/validation_middleware.js

```diff
@@ -0,0 +1,15 @@
+const HttpCodes = require('./httpcodes')
+
+const validationMiddleware = (schema) => {
+    return (request, response, next) => {
+        const {error} = schema.validate(request.body)
+        if (error != null) {
```

Comment:
Small style thing, which is not exactly the same in behavior, but covers your use case here too. To improve readability (at least in my opinionated view) you could use `if (!error)`

Reply:
Also, there's a rule in eslint I like, that forces to use the triple comparison operators, which are more strict and usually what developers want. (Since `!=` and `!==` are not the same)

Reply:
We might disagree with `if (!error)` syntax. I find the ! easy to overlook when skimming code. 

Thanks for the triple comparison comment. I need to understand this a bit

Reply:
You'll find a lot of more accurate and detailed explanations on the web, but the short version is that triple comparison considers the types as well and not only the values. 
And in general, I'd say the more explicit you are, fewer bugs will show up in your code.